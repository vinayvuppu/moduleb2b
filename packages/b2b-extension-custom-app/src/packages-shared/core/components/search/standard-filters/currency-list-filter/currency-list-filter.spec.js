import React from 'react';
import { shallow } from 'enzyme';
import {
  CurrencyListFilter,
  currenciesToOptions,
} from './currency-list-filter';

const createProps = custom => ({
  locale: 'de',
  isMulti: false,
  isSearchable: false,
  currencies: {
    EUR: { label: 'Euro', symbol: '€' },
    USD: { label: 'USD', symbol: '$' },
  },

  // SingleFilter's props
  onUpdateFilter: jest.fn(), // is required in `SingleFilter`
  ...custom,
});

describe('rendering', () => {
  describe('SingleFilter', () => {
    let wrapper;
    let props;
    beforeEach(() => {
      props = createProps();
      wrapper = shallow(<CurrencyListFilter {...props} />);
    });
    it('should render a `SingleFilter` component', () => {
      expect(wrapper).toRender('SingleFilter');
    });
    it('should match output tree', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('currenciesToOptions', () => {
  let list;
  beforeEach(() => {
    list = currenciesToOptions({
      EUR: { label: 'Euro', symbol: '€' },
      USD: { label: 'US-Dollar', symbol: '$' },
    });
  });
  it('should return list of select options', () => {
    expect(list).toEqual([
      { value: 'EUR', label: 'EUR - Euro (€)' },
      { value: 'USD', label: 'USD - US-Dollar ($)' },
    ]);
  });
});
