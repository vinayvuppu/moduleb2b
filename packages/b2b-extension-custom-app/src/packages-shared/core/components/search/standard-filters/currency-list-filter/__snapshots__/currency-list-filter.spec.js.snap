// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering SingleFilter should match output tree 1`] = `
<SingleFilter
  onUpdateValue={[MockFunction]}
  renderInput={[Function]}
/>
`;
